models/classes/runner/time/storageFormat/QtiTimeStorageObjectDecodingTrait.php

```diff
@@ -0,0 +1,52 @@
+<?php
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2020 (original work) Open Assessment Technologies SA ;
+ */
+
+namespace oat\taoQtiTest\models\runner\time\storageFormat;
+
+use oat\taoQtiTest\models\runner\time\AdjustmentMap;
+use oat\taoQtiTest\models\runner\time\QtiTimeLine;
+use oat\taoTests\models\runner\time\TimeLine;
+use oat\taoTests\models\runner\time\TimerAdjustmentMapInterface;
+
+trait QtiTimeStorageObjectDecodingTrait
+{
+    private function decodeTimeline(array $decodedData): array
```

Comment:
Method and variable names are a bit confusing - decode decodedData. Maybe we can rename variable to $timeLineData or method to unserializeTimeline or something like that.

Reply:
Done